src/reducers/fileTree.spec.tsx

```diff
@@ -1012,4 +1013,125 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('findRelativePathWithDiff', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const file3 = 'file3.js';
+
+    const entries = [
+      createInternalVersionEntry({
+        ...fakeVersionEntry,
+        filename: file1,
+        mime_category: 'text',
+        path: file1,
+        status: 'M',
+      }),
+      // Note that file2 does not contain any changes.
+      createInternalVersionEntry({
+        ...fakeVersionEntry,
+        filename: file2,
+        mime_category: 'text',
+        path: file2,
+        status: '',
+      }),
+      createInternalVersionEntry({
+        ...fakeVersionEntry,
+        filename: file3,
+        mime_category: 'text',
+        path: file3,
+        status: 'M',
+      }),
+    ];
+    const version = createVersionWithEntries(entries);
+
+    it('gets the next path which has a diff', () => {
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file1,
+          pathList: [file1, file2, file3],
```

Comment:
It's not easy to look at this test and know what the important inputs are. In this case, `status` is an important input. I'd like to see this use a helper that will let you declare only the important inputs to the test.

Here is a quick idea:

```diff
diff --git a/src/reducers/fileTree.spec.tsx b/src/reducers/fileTree.spec.tsx
index 59d6670..bd41867 100644
--- a/src/reducers/fileTree.spec.tsx
+++ b/src/reducers/fileTree.spec.tsx
@@ -17,6 +17,7 @@ import reducer, {
 } from './fileTree';
 import { createInternalVersion, createInternalVersionEntry } from './versions';
 import { getMessageMap } from './linter';
+import { ExternalVersionEntry } from './versions';
 import {
   createFakeExternalLinterResult,
   createFakeThunk,
@@ -1045,11 +1046,37 @@ describe(__filename, () => {
     ];
     const version = createVersionWithEntries(entries);
 
+    const getFakeVersionAndPathList = (
+      entries: ({ path: string } & Partial<ExternalVersionEntry>)[],
+    ) => {
+      const pathList = entries.map((e) => e.path);
+
+      const version = createVersionWithEntries(
+        entries.map((params) =>
+          createInternalVersionEntry({
+            ...fakeVersionEntry,
+            mime_category: 'text',
+            status: 'M',
+            ...params,
+            filename: params.path,
+          }),
+        ),
+      );
+
+      return { pathList, version };
+    };
+
     it('gets the next path which has a diff', () => {
+      const { pathList, version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
+        { path: file2, status: '' },
+        { path: file3, status: 'M' },
+      ]);
+
       expect(
         findRelativePathWithDiff({
           currentPath: file1,
-          pathList: [file1, file2, file3],
+          pathList,
           position: RelativePathPosition.next,
           version,
         }),
```

This makes it easier for the reader to see that `status: ''` gets skipped.

Also, a helper like this would give you more flexibility. You wouldn't have to include `status: ''` in every test. You could easily define multiple paths, etc.

Reply:
Yes, this is a great suggestion, thanks @kumar303!